core/services/keeper/registry_interface.go

```diff
@@ -209,11 +214,11 @@ func (rw *RegistryWrapper) GetConfig(opts *bind.CallOpts) (*RegistryConfig, erro
 	case RegistryVersion_1_0, RegistryVersion_1_1:
 		config, err := rw.contract1_1.GetConfig(opts)
 		if err != nil {
-			return nil, errors.Wrap(err, "failed to get contract config")
+			return nil, fmt.Errorf("%w [%s]: getConfig v1.%d", ErrContractCallFailure, err, rw.Version)
```

Comment:
I like you are trying to format the errors better, but if you go down this path could you make the errors consistent across different functions of `RegistryWrapper` ?

Couple of points:
- errors.ErrorF is a better function to wrap since it preserves the stack trace
- v1.%d would be confusing when we move to 2.x, `rw.Version` is defined as an enum so better to just log it as is without prepending `v1.`

Reply:
I would love to make errors consistent! I threw this one in as an example. Future PRs can have more, or I can do the refactor in this one. :) 

- `errors.Errorf` is not part of the standard library. Is it preferred that I use `github.com/pkg/errors` then?
- Putting the enum raw value into an error string will give little to no context as to which it is at a glance. What if I added a `String` method to the `RegistryVersion` type that will print a readable value for each enum? That's the standard way of doing it.

Reply:
1. Yeah `github.com/pkg/errors` is used extensively in our repo so i think its fine to use it for more functionality than stdlib.
2. Yep makes sense, might be better doing in a separate PR to avoid snowballing changes